### ISSUE TITLE
login won't work API-key does

### ISSUE THREAD

**Issue by theoneandonly-vector** [tristandostaler/EmpireGUI#1]
when I try to login with my given creds, it only gives the following output:

> https://127.0.0.1:1337/api/admin/login
> POST
> {"username":"empireadmin", "password":"myPW"}
> java.lang.Exception: Invalid PSE connection informations
>     at pse_gui.PowershellEmpireConnection.sendRequest(PowershellEmpireConnection.java:198)
>     at pse_gui.PowershellEmpireConnection.refreshToken(PowershellEmpireConnection.java:114)
>     at pse_gui.PowershellEmpireConnection.makeConnection(PowershellEmpireConnection.java:104)
>     at pse_gui.PowershellEmpireConnection.<init>(PowershellEmpireConnection.java:65)
>     at pse_gui.RequestHandler.connect(RequestHandler.java:19)
>     at pse_gui.MainView$7$1.call(MainView.java:331)
>     at pse_gui.MainView$7$1.call(MainView.java:1)
>     at javafx.concurrent.Task$TaskCallable.call(Task.java:1423)
>     at java.util.concurrent.FutureTask.run(FutureTask.java:266)
>     at javafx.concurrent.Service.lambda$null$492(Service.java:725)
>     at java.security.AccessController.doPrivileged(Native Method)
>     at javafx.concurrent.Service.lambda$executeTask$493(Service.java:724)
>     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
>     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
>     at java.lang.Thread.run(Thread.java:745)
> 
> Login requested..
> user: empieradmin
> host: 127.0.0.1
> port: 1337

and the menus on the left side won't load, so I cant do anything

with the API-key it works flawlessly.
but it will be easier, to use everyday the same login instead of a new API-key.

**Comment by tristandostaler** [comment]
Yes I know. The problem comes from Empire:

https://github.com/adaptivethreat/Empire/issues/331

But thanks!

And I made the token feature just because there was a problem with Empire ahaha

I will close this issue now but if there is (or was) another problem, feel free to post another issue.

Thanks!

**Comment by tristandostaler** [comment]
I added a few features in the dev branch but they are not quite done. But for one of them, I request the server config and in the returned string, accessible in the output window, you have the empire password. So you could use the token once to get the password, then logout and relogin with the retreived password and it would now always work with username/pass.

**Comment by theoneandonly-vector** [comment]
sounds great :)
i'm going to try it out tonight, thx.

Am 24.10.2016 04:53 schrieb "tristandostaler" notifications@github.com:

**Comment by tristandostaler** [comment]
Use the new version released, I added a lot of features (including the part with the config from the server)!

Actually, the dev branch and the master branch are now the same (they have been merged with the new features)

**Comment by theoneandonly-vector** [comment]
jup I already saw that :)
thx, it's now working fine with the login.

Am 25.10.2016 8:40 nachm. schrieb "tristandostaler" <
notifications@github.com>:

**Comment by theoneandonly-vector** [comment]
btw.
are you going to make it working with the v2.0 beta?

**Comment by tristandostaler** [comment]
Yup, it's already in the todo list. The TODO's are in the MainView.java just before the constructor.
The problem with the 2.0 is that the REST API is simply not working so I can't start working on it until it is working. Even making simple requests using curl don't work.
Oh and the EmpireGUI is at version 1.2 and not 2.0 because I want the first digit to follow Empire. So when it will be working with Empire 2, I will make EmpireGUI 2.x. I will also make sure it is retrocompatible, if it is not too hard.

(ps: some features are working, but because of the new architectural design, some features are not, like the modules if I am not mistaking)
